tools/rebuild-transfer-backlog

```diff
@@ -0,0 +1,166 @@
+    try:
+        elasticSearchFunctions.index_files(es_client, 'transfers', 'transferfile', uuid, path, status='backlog')
+    except AttributeError:
+        elasticSearchFunctions.connect_and_index_files('transfers', 'transferfile', uuid, path)  # AM 1.5
+        elasticSearchFunctions.connect_and_change_transfer_file_status(uuid, 'backlog')  # AM 1.5
+
+
+def main():
+    # Make sure the user knows what he or she is going to do
+    print('WARNING: This script will delete your current ElasticSearch transfer data, rebuilding it using files.')
+    proceed = raw_input('Are you sure you want to continue? (yes/no)\n')
+    if proceed != 'yes':
+        print("You didn't enter 'yes': exiting.")
+        sys.exit(0)
+
+    # Determine root of shared directories
+    clientConfigFilePath = '/etc/archivematica/MCPClient/clientConfig.conf'
+    config = ConfigParser.SafeConfigParser()
+    config.read(clientConfigFilePath)
+
+    try:
+        sharedDirectory = config.get('MCPClient', 'sharedDirectoryMounted')
+    except:
+        print('Configuration item "sharedDirectoryMounted" not available at /etc/archivematica/MCPClient/clientConfig.conf.')
+        sys.exit(1)
+
+    # Set transfer backlog directory
+    try:
+        transfer_backlog_dir = sys.argv[1]
+        if not os.path.exists(transfer_backlog_dir):
+            print("Transfer Backlog location doesn't exist.")
+            sys.exit(1)
+    except IndexError:
+        print('Usage: am rebuild-transfer-backlog <Full path to Transfer Backlog>')
+        sys.exit(1)
+
+    print('Rebuilding transfer index from transfer backlog in', transfer_backlog_dir)
+
+    # Verify ES is accessible
+    try:
+        elasticSearchFunctions.setup_reading_from_client_conf()
+        es_client = elasticSearchFunctions.get_client()
+    except AttributeError:
+        es_client = elasticSearchFunctions.Elasticsearch(hosts=elasticSearchFunctions.getElasticsearchServerHostAndPort())  # AM 1.5
+
+    try:
+        es_client.info()
+    except Exception:
+        print("Error: Elasticsearch may not be running.")
+        sys.exit(1)
+
+    print('Deleting all transfers in the "transfers" index')
+    time.sleep(3)  # Time for the user to panic and kill the process
+    es_client.indices.delete('transfers', ignore=404)
+    try:
+        elasticSearchFunctions.create_indexes_if_needed(es_client)
+    except AttributeError:
+        elasticSearchFunctions.check_server_status_and_create_indexes_if_needed()  # AM 1.5
+
+    for directory in os.listdir(transfer_backlog_dir):
+        if directory == '.gitignore':
+            continue
+        process_transfer(es_client, os.path.join(transfer_backlog_dir, directory))
+
+        # TODO: add method to storageService module
+        print("Requesting transfer reindex in Storage Service...")
+        transfer_uuid = directory[-36:]
+        ss_api = storageService._storage_api()
+        ss_api.file(transfer_uuid).reindex().post()
```

Comment:
This will be affected by artefactual/archivematica#550 in AM 1.7.  No change needed now, but we'll need to remember to update this (and handle both versions)